src/futils.cpp

```diff
@@ -71,9 +76,9 @@ namespace Exiv2 {
         return isdigit(ch) ? ch - '0' : tolower(ch) - 'a' + 10;
     } // from_hex
 
-    char* urlencode(char* str) {
-        char* pstr = str;
-        char* buf  = (char*)malloc(strlen(str) * 3 + 1);
+    std::string urlencode(const char* str) {
+        const char* pstr = str;
+        char* buf  = new char[strlen(str) * 3 + 1];
```

Comment:
Depending on your level of motivation, you could change this to the `std::string` as creating it further below causes a copy of the whole string. Instead `buf` itself could be an instance of `std::string` constructed with the correct length.

Reply:
Honestly, I am not very motivated to do it :P .

I changed the signature of the function because I saw some value on using more c++ types and removing from the client code the responsibility of freeing the C-strings. 

But I would not spend more time is this code unless it is strictly necessary. There are other more important things to do in the code base :)

Reply:
Fair enough. Can you add a `\todo` somewhere so that it is not forgotten?